File: 02 백준/04 탐욕법(그리디)/Java/src/Prob11399.java

/**
* title  : 11399. ATM (Silver 4)
* time   : 132ms
* memory : 14420KB
*/

import java.io.*;
import java.util.*;

public class Prob11399 {

    public static void main(String[] args) throws IOException {

        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int N = Integer.parseInt(br.readLine());
        int[] atm = new int[N];

        StringTokenizer st = new StringTokenizer(br.readLine());
        for (int i = 0; i < N; i++) {
            atm[i] = Integer.parseInt(st.nextToken());
        }

        Arrays.sort(atm);

        int answer = 0;
        for (int i = 0; i < N; i++) {
            answer += (atm[i] * (N - i)); 
        }

        System.out.println(answer);

    } // end main
}

File: 02 백준/04 탐욕법(그리디)/Python/Prob11399.py

"""
title  : 11399. ATM (Silver 4)
time   : 52ms       116ms
memory : 31256KB    113112KB
"""

N = int(input())                        # 사람 수
times = list(map(int, input().split())) # 각 사람이 인출하는데 걸리는 시간

answer = 0

times.sort()    # 제일 작은 시간을 최대로 더하고, 제일 큰 시간을 최소로 더하고

for time in times:
    answer += (time * N)
    N -= 1

print(answer)